plugins/woocommerce-blocks/assets/js/blocks/product-template/edit.tsx

```diff
@@ -240,8 +235,29 @@ const ProductTemplateEdit = (
 			}
 			// If `inherit` is truthy, adjust conditionally the query to create a better preview.
 			if ( inherit ) {
-				if ( templateCategory ) {
-					query.categories = templateCategory[ 0 ]?.id;
+				const { taxonomy, slug } = parseTemplateSlug( templateSlug );
+
+				if ( taxonomy && slug ) {
+					const templateCategory = getEntityRecords(
```

Comment:
Nitpick: The name `templateCategory` is a bit misleading. Maybe it can be replaced with `fetchedTaxonomyTerm` OR `taxonomyTermRecord`?